### PR TITLE
fix: Module Debugger error CE change and setter type fix

### LINKED ISSUE #30872: [Bug]: Incorrect setter method params type
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Screenshot 2024-02-01 at 1 14 06 PM

### Steps To Reproduce

1. Go to the application 
2. Add RadioGroup and button widget
3. Write binding `{{RadioGroupd.setData(<<FOCUS HERE>>)}}` 

### Public Sample App

_No response_

### Environment

Production

### Issue video log

_No response_

### Version

1.9.61

### PR DESCRIPTION
**Pull request by rishabhrathod01**
## Description

- Module Instance Debugger CE fixes
- fix for setter method params type being incorrect example:- `RadioGrp1.setData(value:string)` instead of `RadioGrp1.setData(value:[?])` as we default to string when type was array.

#### PR fixes following issue(s)
Fixes #30872 

#### Type of change

- Bug fix (non-breaking change which fixes an issue)

- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **New Features**
	- Enhanced type conversion functionality to support more types, improving the development experience.
- **Improvements**
	- Added an optional `entityName` field for better entity identification.
	- Updated error messages for plugin actions to be more descriptive.
- **Bug Fixes**
	- Fixed data type inconsistencies across various widgets to ensure uniformity in data handling.
- **Refactor**
	- Refactored certain sagas for improved code maintainability and readability.
- **Tests**
	- Adjusted widget tests to reflect the changes in data type handling.

### CODE REVIEW

File: app/client/src/widgets/TableWidgetV2/widget/index.tsx [hunk 1]
```diff
@@ -550,7 +550,7 @@ class TableWidgetV2 extends BaseWidget<TableWidgetProps, WidgetState> {
         },
         setData: {
           path: "tableData",
-          type: "object",
+          type: "array",
```
[the remark this answers is not in the document]
Comment (rishabhrathod01) [reply app/client/src/widgets/TableWidgetV2/widget/index.tsx:553]: It is expeced to be an array.

**Review by ApekshaBhosale — DISMISSED** [review]

**Comment by rohan-arthur** [comment]
this will need a counterpart EE PR

File: app/client/src/ce/utils/actionExecutionUtils.ts [hunk 2]
```diff
@@ -22,6 +22,13 @@ export function getJSActionNameToDisplay(action: JSAction) {
   return action.name;
 }
 
+export function getCollectionNameToDisplay(
+  _: JSAction,
```
Comment (ApekshaBhosale) [inline app/client/src/ce/utils/actionExecutionUtils.ts:26]: this is gonna throw warning during build so please add typescript comment

**Review by ApekshaBhosale — CHANGES_REQUESTED** [review]

File: app/client/src/ce/utils/actionExecutionUtils.ts (hunk 2 above)
Comment (rishabhrathod01) [reply app/client/src/ce/utils/actionExecutionUtils.ts:26]: @ApekshaBhosale we don't get type error for unused param when the param name is `_` , that's the reason for the param name being `_`. Also, the CI client build has passed for the PR.

**Review by ApekshaBhosale — APPROVED** [review]

File: app/client/src/ce/utils/actionExecutionUtils.ts (hunk 2 above)
Comment (rishabhrathod01) [reply app/client/src/ce/utils/actionExecutionUtils.ts:26]: more info :- https://theappsmith.slack.com/archives/C04GRCN4CS3/p1707292451356859?thread_ts=1706092979.859589&cid=C04GRCN4CS3

**Pull request merged**